node/cn/api.go

```diff
@@ -215,18 +215,20 @@ func NewPublicDebugAPI(cn *CN) *PublicDebugAPI {
 }
 
 // DumpBlock retrieves the entire state of the database at a given block.
-func (api *PublicDebugAPI) DumpBlock(blockNr rpc.BlockNumber) (state.Dump, error) {
-	if blockNr == rpc.PendingBlockNumber {
+func (api *PublicDebugAPI) DumpBlock(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) (state.Dump, error) {
+	if *blockNrOrHash.BlockNumber == rpc.PendingBlockNumber {
 		return state.Dump{}, kerrors.ErrPendingBlockNotSupported
 	}
 	var block *types.Block
-	if blockNr == rpc.LatestBlockNumber {
-		block = api.cn.blockchain.CurrentBlock()
+	var err error
+	if *blockNrOrHash.BlockNumber == rpc.LatestBlockNumber {
+		block = api.cn.APIBackend.CurrentBlock()
 	} else {
-		block = api.cn.blockchain.GetBlockByNumber(uint64(blockNr))
+		block, err = api.cn.APIBackend.BlockByNumberOrHash(ctx, blockNrOrHash)
 	}
-	if block == nil {
-		return state.Dump{}, fmt.Errorf("block #%d not found", blockNr)
+	if err != nil {
+		blockNrOrHashString, _ := blockNrOrHash.NumberOrHashString()
+		return state.Dump{}, fmt.Errorf("block %v not found", blockNrOrHashString)
```

Comment:
Why did you handle error out of `else` branch?

Reply:
Oh,, It's my mistake. I moved error check logic in else branch 
Thanks Aidan!